python/pyspark/testing/utils.py

```diff
@@ -115,6 +115,36 @@ def eventually(condition, timeout=30.0, catch_assertions=False):
         )
 
 
+def retry(maxTries=10, interval=1.0):
```

Comment:
we have a similar utility `eventually` at `pyspark.testing.utils`.

Reply:
I see, I mentioned it in the PR description.
I think it will be more easy to use if we make it a decorator

Reply:
I see. Can you make `eventually` to work as a decorator too?

Reply:
sound good, let me have a try

Reply:
it is a bit difference that `eventually` requires the function return `True` if succeed, so I will make it a different decorator

Reply:
Can we reuse `eventually` here?

Reply:
hmm, maybe we can add an additional parameter to treat returned `None` as `True`